src/libpriv/rpmostree-core.c

```diff
@@ -2171,6 +2062,21 @@ rpmostree_context_prepare (RpmOstreeContext *self,
         }
     }
 
+  /* And lock all the base packages we don't expect to be replaced. */
+  { hy_autoquery HyQuery query = hy_query_create (sack);
+    hy_query_filter (query, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
+    g_autoptr(GPtrArray) pkgs = hy_query_run (query);
+    for (guint i = 0; i < pkgs->len; i++)
+      {
+        DnfPackage *pkg = pkgs->pdata[i];
+        const char *pkgname = dnf_package_get_name (pkg);
```

Comment:
Same comment as above regarding error result.

Reply:
Yeah, we usually don't second-guess those `dnf_package_get_*` functions, but we probably should.